pymc3/distributions/continuous.py

```diff
@@ -3836,28 +3834,24 @@ class Interpolated(BoundedContinuous):
         Probability density function evaluated on lattice ``x_points``
     """
 
-    def __init__(self, x_points, pdf_points, *args, **kwargs):
-        self.lower = lower = at.as_tensor_variable(x_points[0])
-        self.upper = upper = at.as_tensor_variable(x_points[-1])
+    rv_op = interpolated
 
-        super().__init__(lower=lower, upper=upper, *args, **kwargs)
+    @classmethod
+    def dist(cls, x_points, pdf_points, *args, **kwargs):
+        lower = at.as_tensor_variable(x_points[0])
+        upper = at.as_tensor_variable(x_points[-1])
 
-        interp = InterpolatedUnivariateSpline(x_points, pdf_points, k=1, ext="zeros")
         Z = interp.integral(x_points[0], x_points[-1])
```

Comment:
Z is not being passed to `super().dist()` and as such won't reach the `logp`. It seems it is not necessary for the random draws so it's probably better to also specify it inside the `logp` from the `x_points` input.